### PR TITLE
Remove fixed data from machine functions

### PR DESCRIPTION
**Pull request by georgwiese**
Most machines that need access to a `FixedData` reference already store it in `self`, so there is no need to pass it in `Machine::process_lookup()` and `Machine::take_witness_col_values()`.

### CODE REVIEW

File: executor/src/witgen/machines/mod.rs [hunk 1]
```diff
@@ -42,14 +41,14 @@ pub trait Machine<'a, T: FieldElement>: Send + Sync {
...
-    fn take_witness_col_values(&mut self, fixed_data: &'a FixedData<T>) -> HashMap<String, Vec<T>>;
+    fn take_witness_col_values(&mut self) -> HashMap<String, Vec<T>>;
 }
 
 /// All known implementations of [Machine].
 /// This allows us to treat machines uniformly without putting them into a `Box`,
 /// which requires that all lifetime parameters are 'static.
 pub enum KnownMachine<'a, T: FieldElement> {
-    SortedWitnesses(SortedWitnesses<T>),
+    SortedWitnesses(SortedWitnesses<'a, T>),
```
Comment (leonardoalt) [inline executor/src/witgen/machines/mod.rs:51]: why is this needed now?

**Review by leonardoalt — APPROVED** [review]
This is probably fine, but I still think that conceptually we should get rid of the idea of "fixed data", fixed columns etc, and just have witnesses

File: executor/src/witgen/machines/mod.rs (hunk 1 above)
Comment (georgwiese) [reply executor/src/witgen/machines/mod.rs:51]: This machine was the only one that actually used fixed data but didn't already store a reference to it. Now, it stores a reference of type `&'a FixedData<'a, T>` (like other machines), so it needs a lifetime parameter itself.
